src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py

```diff
@@ -101,7 +101,10 @@ def get_hostname_completion_list(prefix, action, parsed_args, **kwargs):  # pyli
     register_cli_argument(scope + ' deployment source', 'repo_url', options_list=('--repo-url', '-u'), help='repository url to pull the latest source from, e.g. https://github.com/foo/foo-web')
     register_cli_argument(scope + ' deployment source', 'branch', help='the branch name of the repository')
     register_cli_argument(scope + ' deployment source', 'cd_provider', help='type of CI/CD provider', default='kudu', **enum_choice_list(['kudu', 'vsts']))
-    register_cli_argument(scope + ' deployment source', 'cd_app_type', arg_group='VSTS CD Provider', help='web application framework you used to develop your app', default='AspNetWap', **enum_choice_list(['AspNetWap', 'AspNetCore', 'NodeJSWithGulp', 'NodeJSWithGrunt']))
+    register_cli_argument(scope + ' deployment source', 'cd_app_type', arg_group='VSTS CD Provider', help='web application framework you used to develop your app', default='AspNetWap', **enum_choice_list(['AspNetWap', 'AspNetCore', 'NodeJS', 'PHP', 'Python']))
+    register_cli_argument(scope + ' deployment source', 'nodejs_task_runner', arg_group='VSTS CD Provider', help='Task runner for nodejs. Default : None', default='None', **enum_choice_list(['None', 'Gulp', 'Grunt']))
+    register_cli_argument(scope + ' deployment source', 'python_framework', arg_group='VSTS CD Provider', help='Framework used for Python application. Default : Django', default='Django', **enum_choice_list(['Bottle', 'Django', 'Flask']))
+    register_cli_argument(scope + ' deployment source', 'python_version', arg_group='VSTS CD Provider', help='Python version used for application. Default is 3.5.3x86', default='Python 3.5.3 x86', **enum_choice_list(['Python 2.7.12 x64', 'Python 2.7.12 x86', 'Python 2.7.13 x64', 'Python 2.7.13 x86', 'Python 3.5.3 x64', 'Python 3.5.3 x86', 'Python 3.6.0 x64', 'Python 3.6.0 x86', 'Python 3.6.2 x64', 'Python 3.6.1 x86']))
```

Comment:
It is not flexible to hard-code all these options here in the parameter settings. When there are changes to the options, the settings have to be updated to reflect it. Instead, can this list of options be fetched from other sources, like the SDK? There should be only one source of truth for a piece of information.

Non-blocking.